### ISSUE TITLE
The credentials are not getting considered to get the agents downloaded from appdynamics portal.

### ISSUE THREAD

**Issue by satya66655** [Appdynamics/SampleApp#11]
Hello Team,.

I have been using this script from last two weeks and the credentials which I have been providing in the yml file to login to the appdynamics portal are correct. but it is failing to login when I run the command 
"docker exec -it rest install-appdynamics; docker exec rest start-all"

The error as follows.
[root@apm-test SampleApp]# docker exec -it rest install-appdynamics; docker exec rest start-all
*****************************************************
Using Controller properties from docker-compose.yml
 Controller URL = *******************
 Controller Port = *************
 Account Name = *****************
 Access Key = ******************************
*****************************************************
Downloading Agents from AppDynamics Portal with User: satish.shettar@in.ibm.com
**Incorrect Login/Password**
Environment not set for AppDynamics Agents: exiting...

**Comment by aimrbrto** [comment]
Same problem Here, when I run the docker command :(

**Comment by donhector** [comment]
I too faced the same issue, and I do know for sure my credentials were totally fine. After a little bit of debugging I found several issues with the script `install-appdynamics`. 

1. The `wget` request to get a SSO token is not working as expected.  After sending the request, no key named `sso-sessionid` is found in the cookie, the script tries to extract that key from the cookie and fails miserably with the **Incorrect Login/Password** error. However if I replace this `wget` request:

```
    wget --quiet \
         --save-cookies \
         ${APPD_TEMP_DIR}/cookies.txt \
         --post-data "username=$USER_NAME&password=$PASSWORD" \
         --no-check-certificate \
         -O ${APPD_TEMP_DIR}/index.html \
         ${APPD_LOGIN_URL} 
```

with this `curl` equivalent:

```
curl -sk \
     -c ${APPD_TEMP_DIR}/cookies.txt \
     -d"username=$USER_NAME&password=$PASSWORD" ${APPD_LOGIN_URL}
```

The cookie gets set properly, contains the expected `sso-sessionid` token and the login succeeds.

2. Even once you are logged in successfully, the downloads will fail because they assume that all components (jvm agent, db agent, and machine agent) use the same versions. This no longer seems to valid (maybe it was 2 years ago when this script seems to have been created), so I had to do something like the code below, to provide different versions for the different agents:

```
JVM_VERSION=${JVM_AGENT_VERSION:-"4.5.4.24355"}
DB_VERSION=${DB_AGENT_VERSION:-"4.5.4.885"}
MACHINE_VERSION=${MACHINE_AGENT_VERSION:-"4.5.4.1735"}

APPD_TEMP_DIR=.appd

# Portal Download
APPD_DOWNLOAD_URL=https://aperture.appdynamics.com/download/prox/download-file
APPD_AGENTS=(
  "sun-jvm/${JVM_VERSION}/AppServerAgent-${JVM_VERSION}.zip"
  "db/${DB_VERSION}/dbagent-${DB_VERSION}.zip"
  "machine/${MACHINE_VERSION}/MachineAgent-${MACHINE_VERSION}.zip"
)
```

So after a bit of "hacking" I managed to get the sample up running with docker compose. 

TL;DR:  Script that starts the sample app is obsolete. Appdynamics should fix it ASAP and ensure different versions for different agents can be provided in the `docker-compose.yaml`, specially when they encourage you to use the sample app (https://www.appdynamics.com/sample-app/) during your trial. It does not leave a good impression on potential customers like us to have to fix your code.

**Comment by mgaurav92** [comment]
Hi @donhector 

I was facing the same issue and I replaced 'wget' with 'curl'.
It was working fine previously, but from last few days the response of the curl command is html page with javascript related error.

Can you please help?

```
<!DOCTYPE html>
<html lang="en">
<head>
    <script>
    window["adrum-start-time"] = new Date().getTime();
    window["adrum-config"] = {spa: {angular: {disable: true}},appKey: "AD-AAB-AAB-XTE"};
    </script>
    <script src="lib/adrum/adrum.js"></script>
    <title>AppDynamics </title>

    <!-- Request the best possible renderer in IE -->
    <meta http-equiv="X-UA-Compatible" content="IE=edge"/>
    <meta charset="utf-8"/>

    <!-- TODO: Move font-awesome to build.xml after fixing font-location relative path -->
    <link rel="stylesheet" type="text/css" href="lib/font-awesome/css/font-awesome.min.css?992f3675efe710be0a53fcd7880061be"/>
    <link rel="stylesheet" type="text/css" href="css/css-lib-head-concat.min.css?992f3675efe710be0a53fcd7880061be"/>
    <link rel="icon" href="favicon.ico" type="image/png"/>

    <script defer>
        // UIPLATF-6204 - redirect to /controller if URL doesn't have it
        if (navigator && navigator.userAgent &&
            window.location.pathname && window.location.pathname.indexOf("controller") === -1) {
            window.location.pathname = "/controller/";
        }
    </script>

    <!-- JS files go here, after CSS (chrome suggests that optimization) -->
    <script defer type="text/javascript" src="js-lib-head-concat.js?992f3675efe710be0a53fcd7880061be"></script>
        <!-- webpack prod -->
    <script defer src="dist/main/Bootstrap.webpack.min.js?992f3675efe710be0a53fcd7880061be" charset="utf-8"></script>
    <script defer src="dist/main/vendors.webpack.min.js?992f3675efe710be0a53fcd7880061be" charset="utf-8"></script>
    <script defer src="dist/main/shared.webpack.min.js?992f3675efe710be0a53fcd7880061be" charset="utf-8"></script>
    
    <script defer type="text/javascript" src="singularity.js?992f3675efe710be0a53fcd7880061be"></script>

    <script defer type="text/javascript" src="telemetry.js?992f3675efe710be0a53fcd7880061be"></script>

    <script defer type="text/javascript" src="js-lib-body-concat.js?992f3675efe710be0a53fcd7880061be" charset="utf-8"></script>
    <!-- webpack prod - main module -->
    <script defer src="dist/main/MainAppModuleCode.webpack.min.js?992f3675efe710be0a53fcd7880061be" onload="eval('MainAppModuleCode').default();" charset="utf-8"></script>
    



</head>

<body class="appFrame appBackgroundColor" ng-controller="RouteController">

<!-- This ng-switch will prevent the app from being $compiled until logged in. -->
<div ng-show="visibleItems.appVisible" class="appBackgroundColor appContainer">

    <div class="appContainer" id="appContainer">

        <div class="appFrame appBackgroundColor">

            <!-- ---------------- top menu bar on top of everything: main nav, eval toolbar, etc. ---------------- -->
            <div class="ads-top-menu-bar-container-row ng-cloak" id="topMenuBarContainerDivId">
                <ad-top-menu-bar class="titleBarDiv"></ad-top-menu-bar>
            </div>

            <div ng-class="{'ads-main-container-under-top-menu-bar-minimized':controller.navigationState.isTopMenuBarMinimized,
                                    'ads-main-container-under-top-menu-bar-and-eval-toolbar':controller.navigationState.showEvalToolbar || controller.navigationState.showQuestsToolbar
                                    }" class="ng-cloak ads-main-container-under-top-menu-bar">
                <!-- ---------------- LEFT NAV ---------------- -->
                <!-- adding position: absolute; z-index: 2; to make drop shadow "on top of" right side content in 4.1 -->
                        <div id="leftNavComp" class="leftListDiv leftNavPanel" style="height: 100%; width: 256px; display: inline-block; position: absolute; z-index: 2;">
                    <!-- keep in sync with NAV_PANEL_WIDTH -->
                    <ad-left-menu-bar class="absoluteExpanded"></ad-left-menu-bar>
                </div>

                <!-- ---------------- RIGHT SIDE ---------------- -->

                <div class="ads-right-side-content-container">

                    <!-- ---------------- Page Title / Location Bar / Refresh / Time Range ---------------- -->
                    <div id="mainTitleBarContainerId" class="titleBarDiv">
                        <ad-main-title-bar></ad-main-title-bar>
                    </div>

                    <div class="ads-right-side-under-location-bar-container" ng-switch="loadingContent">

                        <!-- ---------------- Main View Stack / Page Content ---------------- -->
                        <ad-view-stack ad-current-route="currentRoute" ng-switch-when="false">
                        </ad-view-stack>

                        <!--- loading content -->
                        <div class="ads-absolute-expanded" ng-switch-when="true">
                            <div class="ads-display-table ads-height-100percent ads-width-100percent">
                                <div class="ads-display-table-row">
                                    <div class="ads-display-table-cell ads-vertical-align ads-height-100percent ads-width-100percent"
                                         style="text-align: center;">
                                        <i class="fa fa-2x fa-spinner fa-spin"></i>
                                    </div>
                                </div>
                            </div>
                        </div>

                    </div>

                </div>

            </div>

        </div>

    </div>

    <ad-notification-center ad-is-eval="showEvalToolbar"></ad-notification-center>

</div>

<div ng-show="visibleItems.loginVisible" ng-cloak class="sty-login-view">
    <loginview></loginview>
</div>

<div id="flashDiv" ng-show="visibleItems.appVisible" class="flashDiv" swf="Main_38f1dcf744e3e13f336710bc149cc9ec.swf"></div>

<!-- Loading Screen - App Skeleton -->
<div id="uiBootstrapLoadingViewId"
    ng-if="visibleItems.loaderVisible" style="position: absolute; top: 0px; bottom: 0px; left: 0px; right: 0px; background-color: #FFFFFF; z-index: 99;">

    <!---- Main Nav Skeleton --->
    <div style="position: absolute; top: 0px; left: 0px; right: 0px;  ">

        <div class="ads-main-nav-tabs-container ads-border-bottom"
             style="display: flex; flex-wrap: nowrap; justify-content: left; align-items: center;"
        >
        </div>
    </div>

    <div style="position: absolute; top: 50px; left: 0px; right: 0px; bottom: 0px; background-color: #f1f3f5; ">

        <div style="display: flex; flex-direction: column; height: 100%; ">

            <!-- main title bar -->
            <div class="adsMainTitleBarContainerRow" style="display: flex; align-items: center; padding-left: 24px;">
                <div style="border-radius: 4px; background-color: #e8eaeb; height: 18px; width: 160px; ">
                </div>
            </div>

            <!-- main content area -->
            <div style="padding-left: 24px; padding-right: 24px; padding-bottom: 24px; flex: 1;">
                <div class="ads-nb-main-content-area" style="height: 100%; ">
                    <div style="display: flex; align-items: center; justify-content: center;  height: 100%;">
                        <i class="fa fa-2x fa-spinner fa-spin" style="color: #d0d0d0"></i>
                    </div>
                </div>
            </div>

        </div>

    </div>

</div>

<div ng-show="visibleItems.appVisible && visibleItems.containerAppPage" style="z-index:9;" class="appBackgroundColor appContainer">
    <div class="appContainer" id="appContainer">
        <div class="appFrame appBackgroundColor">
            <c-app-root></c-app-root>
        </div>
    </div>
</div>

<noscript>
    <span><strong>Javascript Required.</strong> Enable Javascript in your browser settings to use AppDynamics</span>
</noscript>

<div style="display:none">
    <iframe id="vc" width="1" height="0" frameborder="0" src=""></iframe>
</div>

<platform-notification-container>
</platform-notification-container>

</body>

</html>
```
